File: ARRAYS/Move All Zeroes to End/answers.java
// User function Template for Java

class Solution {
    void pushZerosToEnd(int[] arr) {
        // code here
        int n = arr.length;
        int idx = 0;
        for(int i=0;i<n;i++){
            if(arr[i]!=0)
            arr[idx++] = arr[i];
        }
        while(idx<n){
            arr[idx++] = 0;
        }
    }
}

File: STRINGS/Non Repeating Character/answers.cpp

class Solution {
  public:
    char nonRepeatingChar(string &s) {
        //  code here
        int n = s.length();
        unordered_map<char,int> mp;
        for(int i=0;i<n;i++){
            mp[s[i]]++;
        }
        for(char c : s){
            if(mp[c]==1) return c;
        }
        return '$';
    }
};
